File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        def linkedListToList(self, head):
            if not head:
                return []
            pointer = head
            sll_list = []
            while pointer:
                sll_list.append(pointer.val)
                pointer = pointer.next
            return sll_list

        def jn(l):
            for i in range(len(l)):
                l[i] = str(l[i])
            return l
        
        l1 = int(''.join(jn(linkedListToList(self,l1)[::-1])))
        l2 = int(''.join(jn(linkedListToList(self,l2)[::-1])))
        
        h = ','.join(list(str(l1 +l2))[::-1])
        i = ListNode(h)
        return i     
        

File: 0002-add-two-numbers/0002-add-two-numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *dummyHead = new ListNode(0);
        ListNode *ptr1 = dummyHead, *ptr2 = l1, *ptr3 = l2;
        int carry = 0;
        
        while (ptr2 || ptr3 || carry) {
            int val = carry;
            
            if (ptr2) {
                val += ptr2->val;
                ptr2 = ptr2->next;
            }
            
            if (ptr3) {
                val += ptr3->val;
                ptr3 = ptr3->next;
            }
            
            carry = val / 10;
            ptr1->next = new ListNode(val % 10);
            ptr1 = ptr1->next;
        }

        return dummyHead->next;
    }
};
